optuna/distributions.py

```diff
@@ -416,43 +397,20 @@ class IntUniformDistribution(BaseDistribution):
     """
 
     def __init__(self, low: int, high: int, step: int = 1) -> None:
-        if low > high:
-            raise ValueError(
-                "The `low` value must be smaller than or equal to the `high` value "
-                "(low={}, high={}).".format(low, high)
-            )
-        if step <= 0:
-            raise ValueError(
-                "The `step` value must be non-zero positive value, but step={}.".format(step)
-            )
-
-        high = _adjust_int_uniform_high(low, high, step)
-
-        self.low = low
-        self.high = high
-        self.step = step
-
-    def to_external_repr(self, param_value_in_internal_repr: float) -> int:
-
-        return int(param_value_in_internal_repr)
-
-    def to_internal_repr(self, param_value_in_external_repr: int) -> float:
+        super().__init__(low=low, high=high, log=False, step=step)
 
-        return float(param_value_in_external_repr)
-
-    def single(self) -> bool:
-
-        if self.low == self.high:
-            return True
-        return (self.high - self.low) < self.step
-
-    def _contains(self, param_value_in_internal_repr: float) -> bool:
+    def _asdict(self) -> Dict:
+        d = copy.deepcopy(self.__dict__)
+        d.pop("log")
+        return d
 
-        value = param_value_in_internal_repr
-        return self.low <= value <= self.high and (value - self.low) % self.step == 0
+    def __repr__(self) -> str:
+        kwargs = ", ".join("{}={}".format(k, v) for k, v in sorted(self._asdict().items()))
+        return "{}({})".format(self.__class__.__name__, kwargs)
 
 
-class IntLogUniformDistribution(BaseDistribution):
+@deprecated("3.0.0", "6.0.0", text=_int_distribution_deprecated_msg)
```

Comment:
There is no implementation of `__repr__` of `IntLogUniformDistribution`. Is it intended?

Reply:
I'm sorry but I don't understand what you mean here. I think `__repr__` is already implemented.

https://github.com/optuna/optuna/pull/3244/files#diff-2ff77057715ff5d7b689aa1ba37562d8a47052b826e0267493bfdd13ac7206e3R450-L508